### PR TITLE
change UI to save request in my-collection

### PR DESCRIPTION
**Pull request by rohanrajpal**
re-utilize collections component to select folder and collections

**checks:**

- [x] save request to root collection
- [x] save request to folder
- [x] save request to nested folder
- [x] replace/overwrite old request
- [x] remember location after making a change in request object (renaming request) / losing mouse focus (search ?)
- ~[ ] persist visual indication on selected location even after losing mouse focus~
- [x] mobile / touch device support

**remarks:**
- [ ] save modal is too crowded:
  - [x] remove search
  - [ ] show request path under collections selector
  - [ ] combine request path with request selector
  - [ ] remove request selector and show requests in collections selector
- [ ] allow users to create / edit / delete collections / folders from save request modal - enable overflow button?
- [ ] row-wrapper trigger change location - move select location action to expand / collapse button?

### CODE REVIEW

**Comment by liyasthomas** [comment]
**deploy preview:** https://deploy-preview-1526--hoppscotch-teams.netlify.app

**steps to reproduce:** attaching screen records

bugs:

- collection's existence is ignored

https://user-images.githubusercontent.com/10395817/109828209-70709e00-7c62-11eb-92ef-84d4d480b54b.mp4

---

- can't replace/overwrite previous request
- can't save request to root collection (which has a folder)

https://user-images.githubusercontent.com/10395817/109828577-c5141900-7c62-11eb-8f41-274b213c144e.mp4

---

- clicking request (in nested folders or elsewhere) loads it to request setup - this shouldn't happen
- incorrectly saves request irrespective of last selected location after making a change in request object

https://user-images.githubusercontent.com/10395817/109829434-8763c000-7c63-11eb-95e8-c66f5410b4e0.mp4

@rohanrajpal

**Comment by liyasthomas** [comment]
@rohanrajpal just realized this feature branch is taken from stale `teams-new-ui` branch. Since there's been breaking changes kindly rebase to current `main` branch.

**Comment by rohanrajpal** [comment]
Thanks for the review, working on it right now

**Comment by rohanrajpal** [comment]
u

> **deploy preview:** https://deploy-preview-1526--hoppscotch-teams.netlify.app
> 
> **steps to reproduce:** (attaching screen records)
> 
> bugs:
> 
> * collection's existence is ignored

Doubt, can requests be saved in collections, or only inside the folders of collections?

If no, then I guess the solution to this problem would be a better error message, right?
If yes, then isn't the concept of folders kinda confusing? since I thought that folders are needed for a request to be saved.

**Comment by liyasthomas** [comment]
Requests can be saved to collections as well as folders inside collection.

File: helpers/teams/utils.js [hunk 1]
```diff
@@ -1,268 +1,294 @@
 import gql from "graphql-tag"
+import apollo from "~/plugins/apollo"
```
Comment (liyasthomas) [inline helpers/teams/utils.js:2]: Unused import.

File: components/collections/collection.vue [hunk 2]
```diff
@@ -156,52 +191,74 @@ export default {
...
-          .catch((error) => {
-            // Error
-            this.$toast.error(this.$t("error_occurred"), {
-              icon: "done",
+            .then((data) => {
+              // Result
+              this.$toast.success(this.$t("deleted"), {
+                icon: "delete",
+              })
+              console.log(data)
```
Comment (liyasthomas) [inline components/collections/collection.vue:252]: remove console logs

File: components/collections/folder.vue [hunk 3]
```diff
@@ -137,25 +152,38 @@ export default {
...
-        
+      if (
+        this.showChildren &&
+        this.collectionsType.type == "team-collections" &&
+        this.folder.requests == undefined
+      ) {
+        team_utils
+          .getCollectionRequests(this.$apollo, this.folder.id)
+          .then((requests) => {
+            console.log(requests)
```
Comment (liyasthomas) [inline components/collections/folder.vue:177]: remove console logs

File: components/collections/folder.vue [hunk 4]
```diff
@@ -165,27 +193,27 @@ export default {
...
-            this.$toast.success(this.$t("deleted"), {
-              icon: "delete",
+          team_utils
+            .deleteCollection(this.$apollo, this.folder.id)
+            .then((data) => {
+              // Result
+              this.$toast.success(this.$t("deleted"), {
+                icon: "delete",
+              })
+              console.log(data)
```
Comment (liyasthomas) [inline components/collections/folder.vue:205]: remove console logs

File: components/collections/index.vue [hunk 5]
```diff
@@ -228,13 +264,16 @@ export default {
   methods: {
     updateTeamCollections() {
       console.log(this.collectionsType)
```
Comment (liyasthomas) [inline components/collections/index.vue:266]: remove console logs

File: components/collections/index.vue [hunk 6]
```diff
@@ -275,31 +314,42 @@ export default {
...
-          .catch((error) => {
-            // Error
-            this.$toast.error(this.$t("error_occurred"), {
-              icon: "done",
+            .then((data) => {
+              // Result
+              this.$toast.success(this.$t("folder_added"), {
+                icon: "done",
+              })
+              console.log(data)
```
Comment (liyasthomas) [inline components/collections/index.vue:339]: remove console logs

**Review by liyasthomas — CHANGES_REQUESTED** [review]

File: components/collections/index.vue (hunk 6 above)
Comment (rohanrajpal) [reply components/collections/index.vue:339]: These console logs were there before as well, should I still remove them?

**Review by liyasthomas — APPROVED** [review]

**Comment by liyasthomas** [comment]
@rohanrajpal updating a team request isn't working. I mean - I can't select an existing request from team. Request selector is hidden for Teams tab in save request modal.

**Comment by rohanrajpal** [comment]
> @rohanrajpal updating a team request isn't working. I mean - I can't select an existing request from team. Request selector is hidden for Teams tab in save request modal.
> 
> [Deploy preview ->](https://deploy-preview-1526--hoppscotch-teams.netlify.app)

Yes, so we discussed that edit a request will come in another PR, right? So I'll add that feature for both teams and personal in the next pr

**Comment by liyasthomas** [comment]
Yeah, I almost forgot. Thanks.

**Pull request merged**